melosys-eessi-app/src/main/java/no/nav/melosys/eessi/security/GenericContextExchangeFilter.java

```diff
@@ -0,0 +1,67 @@
+package no.nav.melosys.eessi.security;
+
+import lombok.RequiredArgsConstructor;
+import no.nav.melosys.eessi.service.sts.RestStsClient;
+import no.nav.security.token.support.client.core.ClientProperties;
+import no.nav.security.token.support.client.core.oauth2.OAuth2AccessTokenService;
+import no.nav.security.token.support.client.spring.ClientConfigurationProperties;
+import org.springframework.http.HttpHeaders;
+import org.springframework.stereotype.Component;
+import org.springframework.web.reactive.function.client.ClientRequest;
+import org.springframework.web.reactive.function.client.ClientResponse;
+import org.springframework.web.reactive.function.client.ExchangeFilterFunction;
+import org.springframework.web.reactive.function.client.ExchangeFunction;
+import reactor.core.publisher.Mono;
+
+import javax.annotation.Nonnull;
+import java.util.Optional;
+
+public abstract class GenericContextExchangeFilter implements ExchangeFilterFunction {
```

Comment:
Denne klassen i api ble lagd av Rune for å håndtere både brukertoken og systemtoken og samtidig abstrahere dette for resten av systemet.
Jeg er ikke overbevist at vi trenger en slik klasse i eessi, iaf. ikke med det navnet

Reply:
Usikker hvorvidt vi kommer til å prioritere oppgaven om å gå fra STS til Azure tokens. I det tilfellet ville vi fått gjenbrukt superklassen igjen. Som nevnt valgt jeg å beholde hierarkiet for å få det så lik API som mulig. Litt lettere å sette seg inn i det hvis man allerede er kjent med det.

Reply:
Fjernet generisk klasse.